ci-operator/step-registry/openshift-logging/deploy-operators/openshift-logging-deploy-operators-commands.sh

```diff
@@ -0,0 +1,523 @@
+        run_command "oc get mcp worker -o yaml"
+        run_command "oc get mc $(oc get mcp/worker --no-headers | awk '{print $2}') -o=jsonpath={.spec.config.storage.files}|jq '.[] | select(.path==\"/var/lib/kubelet/config.json\")'"
+
+        exit 1
+    fi
+    set -e
+}
+
+# From 4.11 on, the marketplace is optional.
+# That means, once the marketplace disabled, its "openshift-marketplace" project will NOT be created as default.
+# But, for OLM, its global namespace still is "openshift-marketplace"(details: https://bugzilla.redhat.com/show_bug.cgi?id=2076878),
+# so we need to create it manually so that optional operator teams' test cases can be run smoothly.
+function check_marketplace () {
+    # caps=`oc get clusterversion version -o=jsonpath="{.status.capabilities.enabledCapabilities}"`
+    # if [[ ${caps} =~ "marketplace" ]]; then
+    #     echo "marketplace installed, skip..."
+    #     return 0
+    # fi
+    ret=0
+    run_command "oc get ns openshift-marketplace" || ret=$?
+    if [[ $ret -eq 0 ]]; then
+        echo "openshift-marketplace project AlreadyExists, skip creating."
+        return 0
+    fi
+
+    cat <<EOF | oc create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    security.openshift.io/scc.podSecurityLabelSync: "false"
+    pod-security.kubernetes.io/enforce: baseline
+    pod-security.kubernetes.io/audit: baseline
+    pod-security.kubernetes.io/warn: baseline
+  name: openshift-marketplace
+EOF
+}
+
+# from OCP 4.15, the OLM is optional, details: https://issues.redhat.com/browse/OCPVE-634
+# since OCP4.18, OLMv1 is a new capability: OperatorLifecycleManagerV1
+function check_olm_capability() {
+    # check if OLMv0 capability is added
+    knownCaps=$(oc get clusterversion version -o=jsonpath="{.status.capabilities.knownCapabilities}")
+    if [[ ${knownCaps} =~ "OperatorLifecycleManager\"," ]]; then
+        echo "knownCapabilities contains OperatorLifecycleManagerv0"
+        # check if OLMv0 capability enabled
+        enabledCaps=$(oc get clusterversion version -o=jsonpath="{.status.capabilities.enabledCapabilities}")
+        if [[ ! ${enabledCaps} =~ "OperatorLifecycleManager\"," ]]; then
+            echo "OperatorLifecycleManagerv0 capability is not enabled, skipping operator installation."
+            return 1 # Return a non-zero exit code
+        fi
+    fi
+    return 0 # Return 0 for success
+}
+
+set_proxy
+# Check for required commands
+command -v jq >/dev/null 2>&1 || { echo >&2 "Error: jq is not installed. Aborting."; exit 1; }
+command -v yq-go >/dev/null 2>&1 || { echo >&2 "Error: yq-go is not installed. Aborting."; exit 1; }
+
+run_command "echo $PATH"
+run_command "ls -l /cli"
+run_command "which oc && oc version -o yaml"
+run_command "oc whoami"
+update_global_auth
+sleep 5
+create_icsp_connected
+
+check_olm_capability
+if [[ $? -ne 0 ]]; then
+    echo "Skipping operator installation due to OLM capability."
+    exit 1
+fi
+
+if [[ ! -z ${index_image} ]]; then
```

Comment:
Can you allow us to deploy operator from redhat-operators?

Reply:
We can set `MULTISTAGE_PARAM_OVERRIDE_LOGGING_INDEX_IMAGE: registry.redhat.io/redhat/redhat-operator-index:v4.y`